corehq/apps/accounting/management/commands/suppress_duplicate_invoices.py

```diff
@@ -0,0 +1,84 @@
+from django.core.management.base import BaseCommand
+from django.db.models import Count
+from corehq.apps.accounting.interface import get_subtotal_and_deduction
+from corehq.apps.accounting.models import Invoice, CreditLine, FeatureType
+from corehq.util.dates import get_first_last_days
+
+class Command(BaseCommand):
+    help = 'Put credit back for duplicate invoices and suppress them'
+
+    def add_arguments(self, parser):
+        parser.add_argument('year', type=int, help='The year of the statement period.')
+        parser.add_argument('month', type=int, help='The month of the statement period.')
+        parser.add_argument('note', type=str,
+                            help='A note to be added when crediting back the amount')
+
+    def handle(self, *args, **kwargs):
+        year = kwargs['year']
+        month = kwargs['month']
+        note = kwargs['note']
+
+        start_date, end_date = get_first_last_days(year, month)
+
+        # Filter the Invoice objects based on the date_start
+        invoices_of_given_date = Invoice.objects.filter(date_start__range=(start_date, end_date))
+
+        # Extract the subscription ids that have duplicate invoices
+        duplicate_subs_ids = invoices_of_given_date.values('subscription').annotate(
+            count=Count('id')).filter(count__gt=1).values_list('subscription', flat=True)
+
+        suppressed_count = 0
+        for sub_id in duplicate_subs_ids:
+            related_invoices = list(Invoice.objects.filter(
+                subscription_id=sub_id, date_start__range=(start_date, end_date)).order_by('-date_created'))
+            for invoice in related_invoices[:-1]:
+                if invoice.balance != invoice.subtotal:
+                    #credit them back
+                    self.revert_invoice_payment(invoice, note)
+                #suppress invoice
+                invoice.is_hidden_to_ops = True
+                print(f"Suppressing invoice {invoice.id} for domain {invoice.subscription.subscriber.domain} ",
+                      end='')
+                invoice.save()
+                suppressed_count += 1
+                print("✓")
+
+        self.stdout.write(self.style.SUCCESS(
+            'Successfully suppressed {} duplicate invoices for Statement Period: {}-{}'
+            .format(suppressed_count, year, month)))
+
+    def revert_invoice_payment(self, invoice, note):
```

Comment:
For my understanding - How did we finalize on the implementation on reverting the invoice payment? Is this documented somewhere? Maybe in a ticket or some place else?

Reply:
Documented [here](https://docs.google.com/document/d/1TY2Io4Y7r7IGIHdShg--sHgiWavB35jkjo3VDsiO1M8/edit)